### PR TITLE
Blog: v20 announcement

### PR DESCRIPTION
**Pull request by shlomi-noach**
Announcing Vitess v20 GA, as part of https://github.com/vitessio/vitess/issues/16009

To be merged on Thu, June 27 upon release.

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Announcement preview: https://deploy-preview-1773--vitess.netlify.app/blog/2024-06-27-announcing-vitess-20/

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+slug: '2024-06-27-announcing-vitess-20'
+tags: ['release', 'Vitess', 'v20', 'MySQL', 'kubernetes', 'operator', 'vreplication', 'multi-tenancy', 'Usability', 'Online DDL']
+title: 'Announcing Vitess 20'
+description: "Vitess 20 is now Generally Available"
+---
+
+
+## Announcing Vitess 20
+
+We're delighted to announce the release of Vitess 20 along with the [version 2.13.0](https://github.com/planetscale/vitess-operator/releases/tag/v2.13.0) of the vitess-operator.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]: Vitess 20 should also be a hyperlink.

Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
We're delighted to announce the release of Vitess 20 along with [version 2.13.0](https://github.com/planetscale/vitess-operator/releases/tag/v2.13.0) of the Vitess Kubernetes Operator.
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+Version 20 focuses on usability and on maturity of existing features, and continues to build on the solid foundation of scalability and performance established in previous versions. Our commitment remains steadfast in providing a powerful, scalable, and reliable solution for your database clustering needs.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
Version 20 focuses on usability and maturity of existing features, and continues to build on the solid foundation of scalability and performance established in previous versions. Our commitment remains steadfast in providing a powerful, scalable, and reliable solution for your database scaling needs.
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+## What's New in Vitess 20
+
+- **Query Compatibility**: enhances DML support including improved query compatibility, Vindex hints, and extended support for various sharded `update` and `delete` operations.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
- **Query Compatibility**: enhanced DML support including improved query compatibility, Vindex hints, and extended support for various sharded `update` and `delete` operations.
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+- **VReplication**: multi-tenant imports support.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
- **VReplication**: multi-tenant imports.
```

Comment (deepthi) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: Is it worth saying `(experimental)` for this? @rohit-nayak-ps WDYT?

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+- **Online DDL**: dropping support for `gh-ost` , improve support for various schema change scenarios.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
- **Online DDL**: dropping support for `gh-ost` , improved support for various schema change scenarios.
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+Advanced delete operations with subqueries and multi-target support are included:
+
+```sql
+DELETE FROM t1 WHERE id IN (SELECT col FROM t2 WHERE foo = 32 AND bar = 43);
+DELETE t1, t3 FROM t1 JOIN t2 ON t1.id = t2.id JOIN t3 ON t1.col = t3.col;
+```
+
+These features provide greater control and efficiency for managing sharded databases. For more details, please refer to the Vitess and MySQL documentation.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]: This isn't really about "managing" sharded _databases_ is it? I'd say it's more about managing sharded _data_.

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+### VReplication: Multi-tenant Imports
+
+Many web-scale applications use a multi-tenant architecture where each tenant has their own database (with identical schemas). There are several challenges with this approach like provisioning and scaling, potentially tens of thousands of databases, and uniformly updating database schemas across them.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
Many web-scale applications use a multi-tenant architecture where each tenant has their own database (with identical schemas). There are several challenges with this approach like provisioning and scaling potentially tens of thousands of databases, and uniformly updating database schemas across them.
```

Comment (deepthi) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: I guess technically it's `schema` not `schemas`, but our audience will understand either version.

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+A sharded Vitess cluster is a great option for such a system with a single logical database serving all tenants. Vitess 20 adds support to import data from such a multi-tenant cluster into a single Vitess Cluster, with new options to the MoveTables workflow. You would run one such workflow for each tenant, with imported tenants being served by the Vitess cluster.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
A sharded Vitess cluster is a great option for such a system with a single logical database serving all tenants. Vitess 20 adds support for importing data from such a multi-tenant cluster into a single Vitess Cluster, with new options to the MoveTables workflow. You would run one such workflow for each tenant, with imported tenants being served by the Vitess cluster.
```

Comment (deepthi) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: I do think it's important to say somewhere that this is experimental. Either up top or here. Preferably right at the top in the first mention of this feature.

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+### Online DDL
+
+Release 20.0 drops support for `gh-ost` for Online DDL, as we continue to invest in `vitess` migrations based on vreplication. The `gh-ost` strategy is still recognized; however:
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
Release 20.0 drops support for `gh-ost` for Online DDL, as we continue to invest in `vitess` migrations based on VReplication. The `gh-ost` strategy is still recognized; however:
```

Comment (deepthi) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: I would move the `gh-ost` part of this section to the end. Highlight new features first before talking about dropped/deprecated stuff.

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+- Vttablet binaries no longer bundle the `gh-ost` binary. The user should provide their own `gh-ost` binary, and supply `vttablet --gh-ost-path`.
+- Vitess no longer tests `gh-ost` in CI/endtoend tests.
+
+Vitess migrations now support `enum` definition reordering. Vitess opts to use `enum`s by alias (their string representation) rather than by ordinal value (the internal integer representation).
+
+Vitess has now better analysis for `INSTANT` DDL scenarios, enabled with the `--prefer-instant-ddl` DDL [strategy flag](https://vitess.io/docs/20.0/user-guides/schema-changes/ddl-strategy-flags/). It is able to predict whether a migration can be fulfilled by the `INSTANT` algorithm and use this algorithm if so.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
Vitess now has better analysis for `INSTANT` DDL scenarios, enabled with the `--prefer-instant-ddl` DDL [strategy flag](https://vitess.io/docs/20.0/user-guides/schema-changes/ddl-strategy-flags/). It is able to predict whether a migration can be fulfilled by the `INSTANT` algorithm and use this algorithm if so.
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+It also improves support for range partitioning migrations, and opts to use direct partitioning queries over Online DDL where appropriate.
+
+VDiffs can now be run on Online DDL Workflows which are still in progress (i.e. not yet cut-over).
+
+### Vitess-operator
+
+Automated and scheduled backups are now available as an experimental feature in v2.13.0. The getting started guide contains a [new guide](https://vitess.io/docs/20.0/user-guides/operating-vitess/backup-and-restore/scheduled-backups/) to learn how to use this new feature.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
Automated and scheduled backups are now available as an experimental feature in v2.13.0. We have added a [new user guide](https://vitess.io/docs/20.0/user-guides/operating-vitess/backup-and-restore/scheduled-backups/) for this feature.
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+## Vitess and the Community
+
+As an open-source project, Vitess thrives on the contributions, insights, and feedback from the community. Your experiences and input are invaluable in shaping the future of Vitess. We encourage you to share your stories and ask questions, on [GitHub](https://github.com/vitessio/vitess) or in our [Slack channel](http://vitess.io/slack).
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
As an open-source project, Vitess thrives on the contributions, insights, and feedback from the community. Your experiences and input are invaluable in shaping the future of Vitess. We encourage you to share your stories and ask questions, on [GitHub](https://github.com/vitessio/vitess) or in our [Slack community](http://vitess.io/slack).
```

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+
+## Getting Started
+
+For a seamless transition to Vitess 20, we highly recommend reviewing the [detailed release notes](https://github.com/vitessio/vitess/blob/main/changelog/20.0/20.0.0/release_notes.md). Additionally, explore [our documentation](https://vitess.io/docs/20.0/) for guides, best practices, and tips to make the most of Vitess 20. Whether you're upgrading from a previous version or integrating Vitess for the first time, our resources are designed to support you every step of the way.
```
Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
For a seamless transition to Vitess 20, we highly recommend reviewing the [detailed release notes](https://github.com/vitessio/vitess/blob/main/changelog/20.0/20.0.0/release_notes.md). Additionally, you can explore [our documentation](https://vitess.io/docs/20.0/) for guides, best practices, and tips to make the most of Vitess 20. Whether you're upgrading from a previous version or running Vitess for the first time, our resources are designed to support you every step of the way.
```

Comment (deepthi) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]: Also hyperlink `Vitess 20` to the release.

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 1 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: fixed

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 4 above)
Comment (rohit-nayak-ps) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: Ah, yes, meant to add it originally. We should.

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 1 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: fixed

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 6 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: fixed

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 4 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: added `(experimental)`

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 8 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: done

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 9 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: `gh-ost` moved to the end.

File: content/en/blog/2024-06-27-announcing-vitess-20.md (hunk 13 above)
Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: done

Comment (shlomi-noach) [reply content/en/blog/2024-06-27-announcing-vitess-20.md]: fixed

File: content/en/blog/2024-06-27-announcing-vitess-20.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+DELETE t1, t3 FROM t1 JOIN t2 ON t1.id = t2.id JOIN t3 ON t1.col = t3.col;
+```
+
+These features provide greater control and efficiency for managing sharded data. For more details, please refer to the Vitess and MySQL documentation.
+
+### VReplication: Multi-tenant Imports (experimental)
+
+Many web-scale applications use a multi-tenant architecture where each tenant has their own database (with identical schemas). There are several challenges with this approach like provisioning and scaling potentially tens of thousands of databases, and uniformly updating database schemas across them.
+
+A sharded Vitess cluster is a great option for such a system with a single logical database serving all tenants. Vitess 20 adds support for importing data from such a multi-tenant cluster into a single Vitess Cluster, with new options to the MoveTables workflow. You would run one such workflow for each tenant, with imported tenants being served by the Vitess cluster.
```
Comment (mattlord) [inline content/en/blog/2024-06-27-announcing-vitess-20.md]:
```suggestion
A sharded Vitess [keyspace](https://vitess.io/docs/concepts/keyspace/) is a great option for such a system with a single logical database serving all tenants. Vitess 20 adds support for importing data from such a multi-tenant setup into a single Vitess [keyspace](https://vitess.io/docs/concepts/keyspace/), with new [`--shards` and `--tenant-id` flags](https://vitess.io/docs/reference/programs/vtctldclient/vtctldclient_movetables/vtctldclient_movetables_create/) for the [MoveTables workflow](https://vitess.io/docs/reference/vreplication/movetables/). You would run one such workflow for each tenant, with imported tenants being served by the Vitess cluster.
```

**Review by frouioui — APPROVED** [review]

**Pull request merged**